app/lib/search/backend.dart

```diff
@@ -76,3 +91,102 @@ class SearchBackend {
 }
 
 String _toUrl(String package) => 'https://pub.dartlang.org/packages/$package';
+
+class SnapshotStorage {
+  final Storage storage;
+  final Bucket bucket;
+  final GZipCodec _gzip = new GZipCodec();
+
+  SnapshotStorage(this.storage, this.bucket);
+
+  Future<SearchSnapshot> fetch() async {
+    final List<BucketEntry> list = await bucket.list().toList();
+    final List<String> names = list
+        .where((entry) => entry.isObject)
+        .map((entry) => entry.name)
+        .toList();
+    if (names.isEmpty) return null;
+    // Select the first entry, because there is chance that later entries were
+    // aborted for some reason, and the most probable correct entry is the first
+    // one.
+    names.sort();
+    final String selected = names.first;
```

Comment:
Since the names are sorted by timestamp, this will effectively pick the oldest one. Why not picking `names.last`?

Also: The cloud storage uploads are atomic: Only if all data got written will the object be created.